backend/src/main/java/ai/giskard/service/ProjectService.java

```diff
@@ -93,7 +120,173 @@ public void delete(Long id) {
+            projectKey = project.getKey();
+        }
+        boolean isConflict = projectRepository.findOneByKey(projectKey).isPresent();
+        List<Feedback> feedbacks = mapper.readValue(locationService.resolvedMetadataPath(pathToTimestampDirectory, Feedback.class.getSimpleName()).toFile(), new TypeReference<>(){} );
+
+        // Get the set of every user that participated to feedbacks (reply or feedback post)
+        Set<String> loginsImportedProject = new TreeSet<>() {
+        };
+        feedbacks.forEach(feedback -> {
+            loginsImportedProject.add(feedback.getUser().getLogin());
+            feedback.getFeedbackReplies().forEach(reply -> loginsImportedProject.add(reply.getUser().getLogin()));
+        });
+
+        // Get the set of every user in the instance of giskard
+        List<User> currentUsers = userRepository.findAllByEnabled(true);
+        Set<String> loginsCurrentInstance = new TreeSet<>();
+        currentUsers.forEach(user -> loginsCurrentInstance.add(user.getLogin()));
+
+        return new PrepareImportProjectDTO(isConflict, loginsImportedProject, loginsCurrentInstance);
+    }
+
+    public ProjectDTO importProject(Map<String, String> importedUsersToCurrent, String timestampDirectory, String projectKey, String userNameOwner) throws  IOException {
+        Path pathToTimestampDirectory = locationService.resolvedTmpPath().resolve(timestampDirectory);
+
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory())
+            .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        Project project = mapper.readValue(locationService.resolvedMetadataPath(pathToTimestampDirectory, Project.class.getSimpleName()).toFile(), new TypeReference<>(){});
+        List<ProjectModel> models = mapper.readValue(locationService.resolvedMetadataPath(pathToTimestampDirectory, ProjectModel.class.getSimpleName()).toFile(), new TypeReference<>(){} );
+        List<Dataset> datasets = mapper.readValue(locationService.resolvedMetadataPath(pathToTimestampDirectory, Dataset.class.getSimpleName()).toFile(), new TypeReference<>(){} );
+        List<Feedback> feedbacks = mapper.readValue(locationService.resolvedMetadataPath(pathToTimestampDirectory, Feedback.class.getSimpleName()).toFile(), new TypeReference<>(){} );
+        List<TestSuite> testSuites = mapper.readValue(locationService.resolvedMetadataPath(pathToTimestampDirectory, TestSuite.class.getSimpleName()).toFile(), new TypeReference<>(){} );
 
+        Map<Long, Long> mapFormerNewIdModelDataset = new HashMap<>();
+
+        project.setOwner(userRepository.getOneByLogin(userNameOwner));
+        project.setKey(projectKey);
+        Set<User> guestList = new HashSet<>();
+        importedUsersToCurrent.forEach((key, value) -> {
+            if (!value.equals(userNameOwner) && !value.equals("admin"))
```

Comment:
instead of comparing to "admin" you should rather check if the user has admin right (check SecurityUtils), because the login can be different

Reply:
It's the `value` user that should be checked for admin rights, not the current user, isn't it?